### PR TITLE
feat(choose-UI): Add capabilities to choose a different UI - refactor

### PR DESCRIPTION
**Pull request by labdala**
@thewtex @PaulHax

### CODE REVIEW

File: itkwidgets/viewer.py [hunk 1]
```diff
@@ -35,39 +35,57 @@ async def setup(self):
     async def run(self, ctx):
         """ImJoy plugin setup function."""
         global _viewer_count
+        ui = self._init_viewer_kwargs.get('ui', None)
+        if ui=='pydata-sphinx':
+            config = { 'uiMachineOptions': { 'href': 'https://cdn.jsdelivr.net/npm/itk-viewer-bootstrap-ui@0/dist/itk-viewer-bootstrap-ui.es.js', 'export': 'default' }}
+        elif ui=='mui':
+            config = { 'uiMachineOptions': { 'href': 'https://www.jsdelivr.com/package/npm/itk-viewer-material-ui@0/dist/materialUIMachineOptions.js.es.js', 'export': 'default' }}
+        elif ui!='reference':
```
Comment (thewtex) [inline itkwidgets/viewer.py:43]: apply `black`

File: itkwidgets/viewer.py [hunk 2]
```diff
@@ -35,39 +35,57 @@ async def setup(self):
...
         _viewer_count += 1
 
-        data, input_type = self._get_input_data()
         if data is not None:
+            self.set_default_ui_values(itk_viewer)
             render_type = _detect_render_type(data, input_type)
             if render_type is RenderType.IMAGE:
                 await _set_viewer_image(itk_viewer, data)
             elif render_type is RenderType.POINT_SET:
                 await _set_viewer_point_sets(itk_viewer, data)
```
Comment (thewtex) [inline itkwidgets/viewer.py:67]: Remove these lines

File: itkwidgets/viewer.py [hunk 3]
```diff
@@ -35,39 +35,57 @@ async def setup(self):
...
+            fullscreen=False,
+            data = data,
+             #config should be a python data dictionary and can't be a string e.g. 'pydata-sphinx',
+            config = config,
         )
         _viewer_count += 1
 
-        data, input_type = self._get_input_data()
         if data is not None:
+            self.set_default_ui_values(itk_viewer)
```
Comment (thewtex) [inline itkwidgets/viewer.py:62]: Remove

File: itkwidgets/_initialization_params.py [hunk 4]
```diff
@@ -28,4 +28,5 @@ def init_params_dict(itk_viewer):
         'x_slice': itk_viewer.setXSlice,
         'y_slice': itk_viewer.setYSlice,
         'z_slice': itk_viewer.setZSlice,
+        'ui': itk_viewer.setUI,
```
Comment (thewtex) [inline itkwidgets/_initialization_params.py:31]: Remove?

File: itkwidgets/viewer.py [hunk 5]
```diff
@@ -35,39 +35,57 @@ async def setup(self):
...
+        if ui=='pydata-sphinx':
+            config = { 'uiMachineOptions': { 'href': 'https://cdn.jsdelivr.net/npm/itk-viewer-bootstrap-ui@0/dist/itk-viewer-bootstrap-ui.es.js', 'export': 'default' }}
+        elif ui=='mui':
+            config = { 'uiMachineOptions': { 'href': 'https://www.jsdelivr.com/package/npm/itk-viewer-material-ui@0/dist/materialUIMachineOptions.js.es.js', 'export': 'default' }}
+        elif ui!='reference':
+            config=ui
+        else:
+             config={}
+        
+        data, input_type = self._get_input_data()
```
Comment (thewtex) [inline itkwidgets/viewer.py:48]:
```
init_data = None
if data is not None:
            render_type = _detect_render_type(data, input_type)
            if render_type is RenderType.IMAGE:
                data_init = { 'image': data }
            elif render_type is RenderType.POINT_SET:
                data_init = { 'pointSet': data }
```

File: itkwidgets/viewer.py [hunk 6]
```diff
@@ -35,39 +35,57 @@ async def setup(self):
...
+        
         itk_viewer = await api.createWindow(
             name =f'itkwidgets viewer {_viewer_count}',
             type='itk-vtk-viewer',
-            src='https://kitware.github.io/itk-vtk-viewer/app',
-            fullscreen=True,
+            src= 'http://localhost:8082/', #for testing purposes
+            fullscreen=False,
+            data = data,
```
Comment (thewtex) [inline itkwidgets/viewer.py:55]:
```
data = data_init
```

**Review by thewtex — CHANGES_REQUESTED** [review]

**Comment by labdala** [comment]
It is responsive now, thank you so so much @thewtex and @bnmajor

**Comment by labdala** [comment]
Addresses changes requested in #477.

**Review by thewtex — APPROVED** [review]
:tada: :taco:

**Pull request merged**